rsl/rsl.go

```diff
@@ -39,37 +43,132 @@ func InitializeNamespace() error {
 	return nil
 }
 
-// AddEntry adds an RSL entry to the log for the ref and hash passed in.
-func AddEntry(refName string, commitID plumbing.Hash, sign bool) error {
+type RSLEntry struct {
+	RefName  string
+	CommitID plumbing.Hash
+	Revoke   []plumbing.Hash
+}
+
+// NewNormalEntry returns an RSLEntry object for a normal RSL entry.
+// TODO: fix docstring
+func NewNormalEntry(refName string, commitID plumbing.Hash) *RSLEntry {
+	return &RSLEntry{RefName: refName, CommitID: commitID, Revoke: []plumbing.Hash{}}
+}
+
+// NewRevocationEntry returns an RSLEntry object that revokes a prior, specified
+// RSL entry.
+// TODO: fix docstring
+func NewRevocationEntry(refName string, commitID plumbing.Hash, revokedEntryIDs []plumbing.Hash) *RSLEntry {
+	return &RSLEntry{RefName: refName, CommitID: commitID, Revoke: revokedEntryIDs}
+}
```

Comment:
up to you: If you want you could use a variadic param here:

```suggestion
func NewRevocationEntry(refName string, commitID plumbing.Hash, revokedEntryIDs ...plumbing.Hash) *RSLEntry {
	return &RSLEntry{RefName: refName, CommitID: commitID, Revoke: revokedEntryIDs}
}
```

Reply:
Good call, this is neater.

Reply:
[3fbfb93](https://github.com/adityasaky/gittuf/pull/24/commits/3fbfb93fd65b7ccd8983df4e5c8ceffe41d7a021)